File: packages/core/src/profile/registry.ts
import type { RequirementProfile } from "./types.js";
import { standardProfile } from "./standard.js";

export const STANDARD_PROFILE_ID = "standard";

const profiles: Record<string, RequirementProfile> = {
  [STANDARD_PROFILE_ID]: standardProfile,
};

export function getRequirementProfile(id: string): RequirementProfile {
  const p = profiles[id];
  if (p === undefined) {
    throw new Error(`Unknown requirement profile: ${id}`);
  }
  return p;
}

export function listRegisteredProfileIds(): string[] {
  return Object.keys(profiles);
}

File: crates/gitreqd-core/src/profile/registry.rs
//! GRD-SYS-010: Profile registry.

use crate::error::Error;

use super::standard::StandardProfile;
use super::types::RequirementProfile;

pub const STANDARD_PROFILE_ID: &str = "standard";

static STANDARD: StandardProfile = StandardProfile;

pub fn get_requirement_profile(id: &str) -> Result<&'static dyn RequirementProfile, Error> {
    match id {
        STANDARD_PROFILE_ID => Ok(&STANDARD),
        other => Err(Error::msg(format!("Unknown requirement profile: {other}"))),
    }
}

pub fn list_registered_profile_ids() -> Vec<&'static str> {
    vec![STANDARD_PROFILE_ID]
}
